### PR TITLE
Allow taker to open a CFD

### PR DESCRIPTION
**Pull request by luckysori**
As far as I can tell, the taker is able to open a CFD.

### CODE REVIEW

File: rust/src/lightning.rs [hunk 1]
```diff
@@ -488,28 +454,62 @@ pub fn setup(
...
+        channel_manager: system.channel_manager.clone(),
+        wallet: system.wallet.clone(),
+        network_graph: system.network_graph.clone(),
+        keys_manager: system.keys_manager.clone(),
+        inbound_payments: system.inbound_payments.clone(),
+        outbound_payments: system.outbound_payments.clone(),
+        network: system.network,
+    };
+
+    // Step 17: Initialize routing ProbabilisticScorer
```
Comment (bonomat) [inline rust/src/lightning.rs:478]: Does `Step17` this still make sense?

**Review by da-kami — APPROVED** [review]
Noice! 

@luckysori it would be sweet if you could fix the copy-pasters I introduced before merging 😬

The TODOs related to persisting CFD state on taker and maker were captured in:

https://github.com/itchysats/10101/issues/201
https://github.com/itchysats/10101/issues/203

File: lib/cfd_trading/cfd_order_confirmation.dart [hunk 2]
```diff
@@ -66,24 +70,39 @@ class CfdOrderConfirmation extends StatelessWidget {
...
-                                // todo send order to maker.
+
+                                // TODO: Plug in actual state changes
 
                                 // clear draft order from cfd service state
                                 cfdTradingService.draftOrder = null;
 
+                                try {
+                                  // TODO: Don't hardcode the values
+                                  await api.openCfd(takerAmount: 20000, leverage: 2);
```
Comment (da-kami) [inline lib/cfd_trading/cfd_order_confirmation.dart:84]: Not sure if we should keep the hardcoded value or use the actual margin. I would vote for using the actual margin but keep the leverage locked in at 2 until we action #198 

I also captured https://github.com/itchysats/10101/issues/200 which is related.

@luckysori if we keep the hardcoded value please add a comment to https://github.com/itchysats/10101/issues/200 :)

Comment (luckysori) [reply lib/cfd_trading/cfd_order_confirmation.dart:84]: Let's keep it hard-coded until we action #200. I'll add the comment.

**Comment by da-kami** [comment]
resolves https://github.com/itchysats/10101/issues/182

(follow up tickets captured)

**Pull request merged**